### DISCUSSION TITLE
mini diff - way to programmatically check if it's enabled and if the overlay is enabled

### DISCUSSION THREAD

**Discussion by beaumccartney** [nvim-mini/mini.nvim#1060] [Q&A] [answered]
I'm looking to make toggle maps in the style of mini.basics toggle diagnostics, where i'd like to print the end state of the toggle after. is there an api or other way to check if mini.diff is enabled and if the overlay is enabled?

**Comment by echasnovski** [comment, marked as answer]
Yes, there is. Take a look at [`get_buf_data()`](https://github.com/echasnovski/mini.nvim/blob/eb19e8442edd63b033c36daf8d29c0158c9f1eb9/doc/mini-diff.txt#L478-L495):
- It [returns `nil` if buffer is disabled](https://github.com/echasnovski/mini.nvim/blob/eb19e8442edd63b033c36daf8d29c0158c9f1eb9/doc/mini-diff.txt#L486) and a table otherwise.
- For enabled buffers with attached source the table has [`overlay` field](https://github.com/echasnovski/mini.nvim/blob/eb19e8442edd63b033c36daf8d29c0158c9f1eb9/doc/mini-diff.txt#L490) showing whether an overlay is active.

**Reply by beaumccartney** [reply]
Thanks!
